src/display.cpp

```diff
@@ -3309,7 +3315,7 @@ void display::process_reachmap_changes()
 
 		for (const auto& hex : get_visible_hexes()) {
 			reach_map::iterator reach = full.find(hex);
-			if (reach == full.end()) {
+			if (reach != full.end()) {
```

Comment:
This change breaks the else below.

Reply:
How has it been broken? I rebuilt master with debug messages in both the `if` and the `else if` and both are working fine:

```
			if (reach != full.end()) {
				DBG_DP << "hex" << hex << "is within reach";
				// Location needs to be darkened or brightened
				invalidate(hex);
			} else if (reach->second != 1) {
				DBG_DP << "hex" << hex << "will be invalidated";
				// Number needs to be displayed or cleared
				invalidate(hex);
			}

```

Resulted in:

>20241029 20:37:39 debug display: hex26,25is within reach
20241029 20:37:39 debug display: hex26,26will be invalidated

It does _seem_ to be redundant, as the unreachable hexes should not be invalidated here, line 3327 below is managing the updating of old and new reach already.

Reply:
You're not allowed to dereference `end()`. That's undefined behavior. If it doesn't crash for you that's just one of the ways undefined behavior can manifest.

Reply:
Oh, I see. I also see it is being managed in #9446, if I'm expected to PR the proposed fix, I will.